packages/asset-swapper/src/utils/market_operation_utils/path_optimizer.ts

```diff
@@ -77,8 +77,15 @@ function findRoutesAndCreateOptimalPath(
     fees: FeeSchedule,
     neonRouterNumSamples: number,
 ): Path | undefined {
-    const createFill = (sample: DexSample) =>
-        dexSamplesToFills(side, [sample], opts.outputAmountPerEth, opts.inputAmountPerEth, fees)[0];
+    const createFill = (sample: DexSample): Fill | undefined => {
+        const fills = dexSamplesToFills(side, [sample], opts.outputAmountPerEth, opts.inputAmountPerEth, fees);
```

Comment:
Won't fills from the same source be disconnected from each other? `dexSamplesToFills()` generates a `sourcePathId` common to all samples passed in. But since you call this sample by sample, they'll have different `sourcePathId`s which might mess with other logic in AS ([example](https://github.com/0xProject/protocol/blob/kim/fix/fix-creating-fills-for-zero-output/packages/asset-swapper/src/utils/market_operation_utils/path.ts#L257)). `index` and `parent` fields will also be disconnected (always `0` and `undefined`). Maybe those were just used during routing though.

Reply:
`index` and `parent` are fine, they are intentionally overridden to `0` and `undefined` when creating the adjusted fills anyway.

The effect on `sourcePathId` is true, good catch! 🙏 The router doesn't care about them, but I'm not sure what the side effect of not merging fills when collapsing. Any ideas?

I think the medium-term plan is to try to move away from the fills/path abstraction into something that is leaner and closer to what we use in the Rust router (lots of unnecessary data-wrangling going on right now with unclear benefit).

Reply:
I think it might possibly cause us to swap on (for example) uniswap 13 times even though it's all the same. :shrug:

Reply:
Well that would be bad :feelsgood: